include/glow/CodeGen/MemoryAllocator.h

```diff
@@ -27,11 +27,44 @@
 
 namespace glow {
 
+/// Utility function to verify if two contiguous integer intervals overlap. Each
+/// interval is described by a half-open interval [begin, end).
+template <class T>
+inline bool intervalsOverlap(T begin1, T end1, T begin2, T end2) {
+  return (std::max(begin1, begin2) < std::min(end1, end2));
+}
+
+/// Allocation structure which represents a request to allocate or free
+/// a memory region (buffer) within a given MemoryAllocator instance.
+struct Allocation {
+
+  /// Type that should be used as a handle.
+  using Handle = const void *;
+
+  /// Allocation handle which uniquely identifies the buffer to be allocated.
+  /// The provided handle must be unique for each buffer.
+  Handle handle_;
+
+  /// Allocation request type flag: true for ALLOC, false for FREE.
+  bool alloc_;
```

Comment:
It could be better to put small data members like this after all other  data members due to alignment. The whole struct will use less space.

Reply:
Done.